src/validation.h

```diff
@@ -306,7 +306,7 @@ void PruneBlockFilesManual(int nManualPruneHeight);
  * plTxnReplaced will be appended to with all transactions replaced from mempool **/
 bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransactionRef &tx,
                         bool* pfMissingInputs, std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, const CAmount nAbsurdFee, bool test_accept=false) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+                        bool bypass_limits, const CAmount nAbsurdFee, bool test_accept=false, bool test_accept_unsigned=false) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
```

Comment:
This already has way too many positional arguments. I'd prefer to make `test_accept` a `enum class` to avoid passing in single bits that are only supposed to be set exclusively?

Reply:
I've replaced the `test_accept` by an `enum class` called `MemPoolValidationScope`. Is this better? Any suggestion is very welcome.